code/parachain/frame/assets-interface/src/lib.rs

```diff
@@ -0,0 +1,344 @@
+			increment::{Increment, SafeIncrement},
+			start_at::ZeroInit,
+		},
+	};
+	pub use composable_traits::{
+		assets::Asset,
+		currency::{
+			AssetExistentialDepositInspect, AssetRatioInspect, BalanceLike, Exponent,
+			Rational64 as Rational,
+		},
+		defi::Ratio,
+		xcm::assets::{ForeignMetadata, RemoteAssetRegistryInspect, RemoteAssetRegistryMutate},
+	};
+
+	pub use codec::{Decode, Encode};
+
+	use composable_traits::{
+		assets::{
+			AssetInfo, AssetInfoUpdate, BiBoundedAssetName, BiBoundedAssetSymbol, GenerateAssetId,
+		},
+		storage::UpdateValue,
+		xcm::assets::{CosmWasmIssuance, PermissionlessAsset},
+	};
+	use frame_support::{
+		dispatch::DispatchResult,
+		pallet_prelude::*,
+		traits::{fungibles::Mutate, Currency, EnsureOrigin, ExistenceRequirement::KeepAlive},
+	};
+	use frame_system::pallet_prelude::*;
+	use scale_info::TypeInfo;
+	use sp_std::fmt::Debug;
+
+	/// The module configuration trait.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The overarching event type.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// Factory to create new lp-token.
+		type AssetId: FullCodec
+			+ Eq
+			+ PartialEq
+			+ Copy
+			+ MaybeSerializeDeserialize
+			+ From<u128>
+			+ Into<u128>
+			+ Debug
+			+ Default
+			+ Ord
+			+ TypeInfo
+			+ MaxEncodedLen;
+
+		/// Foreign location of an asset
+		type Location: FullCodec
+			+ Eq
+			+ PartialEq
+			+ MaybeSerializeDeserialize
+			+ Debug
+			+ Clone
+			+ TypeInfo
+			+ MaxEncodedLen;
+
+		type Balance: BalanceLike + From<u128>;
+
+		type Currency: Currency<Self::AccountId, Balance = Self::Balance>;
+
+		type AssetsRegistry: RemoteAssetRegistryMutate<
+				AssetId = Self::AssetId,
+				AssetNativeLocation = Self::Location,
+				Balance = Self::Balance,
+			> + GenerateAssetId<AssetId = Self::AssetId>
+			+ AssetExistentialDepositInspect<AssetId = Self::AssetId, Balance = Self::Balance>;
+
+		/// Asset creation fee goes to treasury.
+		type TreasuryAccount: Get<Self::AccountId>;
+
+		// Set asset creation fee origin
+		type SetFeeOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		type Assets: Mutate<Self::AccountId, Balance = Self::Balance, AssetId = Self::AssetId>;
+
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	/// Asset Nonce
+	#[pallet::storage]
+	#[allow(clippy::disallowed_types)] // Allow for `ValueQuery` because of nonce
+	pub type AssetNonce<T: Config> =
+		StorageValue<_, u64, ValueQuery, Nonce<ZeroInit, SafeIncrement>>;
+
+	/// Fee to pay for new asset registry
+	#[pallet::storage]
+	pub type AssetCreationFee<T: Config> = StorageValue<_, T::Balance, OptionQuery>;
+
+	/// Asset admin
+	#[pallet::storage]
+	pub type AssetAdmin<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::AssetId, T::AccountId, OptionQuery>;
+
+	/// Is Asset CosmWasm Asset
+	#[pallet::storage]
+	pub type IsCosmWasmAsset<T: Config> =
```

Comment:
IsPermissionlessAsset

Reply:
This is for CosmWasm only to allow mint/burn.